src/sql/src/plan/query.rs

```diff
@@ -1588,15 +1570,48 @@ fn plan_view_select(
         relation_expr = relation_expr.filter(vec![expr]);
     }
 
-    // Step 3. Gather aggregates.
-    let aggregates = {
-        let mut aggregate_visitor = AggregateFuncVisitor::new(&qcx.scx);
-        aggregate_visitor.visit_select(&s);
-        for o in order_by_exprs {
-            aggregate_visitor.visit_order_by_expr(o);
+    // Step 3. Gather aggregates and table functions.
+    let (aggregates, table_funcs) = {
+        let mut visitor = AggregateTableFuncVisitor::new(&qcx.scx);
+        visitor.visit_select_mut(&mut s);
+        for o in order_by_exprs.iter_mut() {
+            visitor.visit_order_by_expr_mut(o);
         }
-        aggregate_visitor.into_result()?
+        visitor.into_result()?
     };
+    let mut table_func_names: HashMap<String, Ident> = HashMap::new();
+    if !table_funcs.is_empty() {
```

Comment:
For clarity of the outer function can we tuck this away somewhere? If you feel that isn't a good idea whatsoever, I'm easily convinced otherwise but this really feels like the sausage-making of this feature.

Reply:
I've stuffed this into a little function. I agree it looks better now even though it's a tad weird.